app/src/main/java/io/appium/uiautomator2/model/internal/CustomUiDevice.java

```diff
@@ -131,26 +130,31 @@ private UiObject2 toUiObject2(@Nullable Object selector, @Nullable Accessibility
     @Nullable
     public AccessibleUiObject findObject(Object selector) throws UiAutomator2Exception {
         final AccessibilityNodeInfo node;
+        final BySelector realSelector;
         if (selector instanceof BySelector) {
             node = (AccessibilityNodeInfo) invoke(METHOD_FIND_MATCH, ByMatcherClass,
                     Device.getUiDevice(), selector, getCachedWindowRoots());
+            realSelector = (BySelector) selector;
         } else if (selector instanceof NodeInfoList) {
             node = ((NodeInfoList) selector).getFirst();
-            selector = toSelector(node);
+            realSelector = toBySelector(node);
         } else if (selector instanceof AccessibilityNodeInfo) {
             node = (AccessibilityNodeInfo) selector;
-            selector = toSelector(node);
+            realSelector = toBySelector(node);
         } else if (selector instanceof UiSelector) {
             return toAccessibleUiObject(getUiDevice().findObject((UiSelector) selector));
         } else {
-            throw new InvalidSelectorException("Selector of type " + selector.getClass().getName() + " not supported");
+            throw new InvalidSelectorException(String.format(
+                    "Selector of type %s not supported",
+                    selector == null ? null : selector.getClass().getName()
+            ));
         }
-        return node == null ? null : new AccessibleUiObject(toUiObject2(selector, node), node);
+        return node == null ? null : new AccessibleUiObject(toUiObject2(realSelector, node), node);
     }
 
     public synchronized GestureController getGestureController() {
         if (gestureController == null) {
-            UiObject2 dummyElement = toUiObject2(null, null);
+            UiObject2 dummyElement = toUiObject2(makeDummySelector(), null);
```

Comment:
Here should not be a dummy selector. For example:

```
[3] pry(main)> e = driver.find_element id: 'com.android.chrome:id/url_bar'
=> #<Appium::Core::Element:0x633a5f9684e8112a id="00000000-0000-0127-ffff-ffff00000026">
[4] pry(main)> e.click
Selenium::WebDriver::Error::UnknownError: An unknown server-side error occurred while processing the command. Original error: Cannot create UiObject2 instance with 'BySelector [RES='\QDUMMY:id/95759079-6f8e-48de-ae7a-2c145fdce875\E']' selector
from io.appium.uiautomator2.common.exceptions.UiAutomator2Exception: Cannot create UiObject2 instance with 'BySelector [RES='\QDUMMY:id/95759079-6f8e-48de-ae7a-2c145fdce875\E']' selector
```

(Open a chrome browser and tap the URL search bar)

```
[AndroidUiautomator2Driver@ac8b (73854af0)] Encountered internal error running command: An unknown server-side error occurred while processing the command. Original error: Cannot create UiObject2 instance with 'BySelector [RES='\QDUMMY:id/95759079-6f8e-48de-ae7a-2c145fdce875\E']' selector
[AndroidUiautomator2Driver@ac8b (73854af0)] io.appium.uiautomator2.common.exceptions.UiAutomator2Exception: Cannot create UiObject2 instance with 'BySelector [RES='\QDUMMY:id/95759079-6f8e-48de-ae7a-2c145fdce875\E']' selector
[AndroidUiautomator2Driver@ac8b (73854af0)] 	at io.appium.uiautomator2.model.internal.CustomUiDevice.toUiObject2(CustomUiDevice.java:121)
[AndroidUiautomator2Driver@ac8b (73854af0)] 	at io.appium.uiautomator2.model.internal.CustomUiDevice.getGestureController(CustomUiDevice.java:157)
[AndroidUiautomator2Driver@ac8b (73854af0)] 	at io.appium.uiautomator2.core.AxNodeInfoHelper.click(AxNodeInfoHelper.java:131)
[AndroidUiautomator2Driver@ac8b (73854af0)] 	at io.appium.uiautomator2.model.BaseElement.click(BaseElement.java:70)
[AndroidUiautomator2Driver@ac8b (73854af0)] 	at io.appium.uiautomator2.handler.Click.safeHandle(Click.java:37)
[AndroidUiautomator2Driver@ac8b (73854af0)] 	at io.appium.uiautomator2.handler.request.SafeRequestHandler.handle(SafeRequestHandler.java:59)
[AndroidUiautomator2Driver@ac8b (73854af0)] 	at io.appium.uiautomator2.server.AppiumServlet.handleRequest(AppiumServlet.java:277)
[AndroidUiautomator2Driver@ac8b (73854af0)] 	at io.appium.uiautomator2.server.AppiumServlet.handleHttpRequest(AppiumServlet.java:271)
[AndroidUiautomator2Driver@ac8b (73854af0)] 	at io.appium.uiautomator2.http.ServerHandler.channelRead(ServerHandler.java:77)
[AndroidUiautomator2Driver@ac8b (73854af0)] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
[AndroidUiautomator2Driver@ac8b (73854af0)] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
[AndroidUiautomator2Driver@ac8b (73854af0)] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
```

Reply:
Or for now, we can use this dummy value with FIXME comment since this works with 2.2.0
(I'm ok with this)

Reply:
thanks for checking it with the newer version. Let me try another trick

Reply:
I've added a TODO there. It looks like the internal `Gestures` class has been significantly changes in v 2.3.0, so the current approach won't work after upgrade anyway